LuaRules/Gadgets/cmd_retreat.lua

```diff
@@ -375,14 +389,21 @@ local function CheckSetWantRetreat(unitID)
 	if not retreatState[unitID] or retreatState[unitID] == 0 then
 		return
 	end
-	
+	local shieldratio
+	local unitdef = spGetUnitDefID(unitID)
+	if shields[unitdef] then
+		local _, currentcharge = spGetUnitShieldState(unitID)
+		shieldratio = currentcharge / shields[unitdef]
```

Comment:
Name (it's not the unit def, it's the ID) and microoptimisation:
```suggestion
	local unitDefID = spGetUnitDefID(unitID)
	local maxShield = shields[unitDefID]
	if maxShield then
		local _, currentcharge = spGetUnitShieldState(unitID)
		shieldratio = currentcharge / maxShield
```